### PR TITLE
Replace cloud provider string with enum

### PR DESCRIPTION
**Pull request by milan-deepfence**
Fixes https://github.com/deepfence/enterprise-roadmap/issues/2199

Changes proposed in this pull request:
- use enum for providers, k8 status

### CODE REVIEW

File: deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/CloudResourcesTable.tsx [hunk 1]
```diff
@@ -101,27 +101,27 @@ const FILTER_SEARCHPARAMS: Record<string, string> = {
...
-  },
-];
-
 const getAppliedFiltersCount = (searchParams: URLSearchParams) => {
   return Object.keys(FILTER_SEARCHPARAMS).reduce((prev, curr) => {
     return prev + searchParams.getAll(curr).length;
   }, 0);
 };
 
+function getCloudResourcePrettyName(provider: ModelCloudResourceCloudProviderEnum) {
```
Comment (manV) [inline deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/CloudResourcesTable.tsx:110]: `getCloudProviderPrettyName` would be a better name

File: deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/PodsTable.tsx [hunk 2]
```diff
@@ -129,10 +129,13 @@ const getAppliedFiltersCount = (searchParams: URLSearchParams) => {
     return prev + searchParams.getAll(curr).length;
   }, 0);
 };
+enum KunernetesEnum {
```
Comment (manV) [inline deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/PodsTable.tsx:132]: I don't think this is any better, we can revert this

**Review by manV — CHANGES_REQUESTED** [review]
minor changes needed

File: deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/PodsTable.tsx (hunk 2 above)
Comment (milan-deepfence) [reply deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/PodsTable.tsx]: fixed

File: deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/CloudResourcesTable.tsx (hunk 1 above)
Comment (milan-deepfence) [reply deepfence_frontend/apps/dashboard/src/features/topology/data-components/tables/CloudResourcesTable.tsx]: fixed

**Pull request merged**